demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+
+    mapping_list = []
+    for (field_name, xdm_field_name) in zip(names_list, xdm_one_data_model_list):
+        type_raw, is_array_raw = extract_raw_type_data(raw_event, field_name)
+        xdm_field_type = xdm_rule_to_dtype.get(xdm_field_name)
+        xdm_class_type = xdm_rule_to_dclass.get(xdm_field_name)
+
+        mapping_list.append(
+            MappingField(
+                xdm_rule=xdm_field_name,
+                field_path_raw=field_name,
+                xdm_field_type=xdm_field_type,
+                xdm_class_type=xdm_class_type,
+                is_array_raw=is_array_raw,
+                type_raw=type_raw,
+            )
+        )
+
+    return mapping_list
+
+
+def convert_raw_type_to_xdm_type(schema_type: str) -> str:
+    """
+    returns the xdm type convention
+    """
+    converting_dict = {"string": "String", "int": "Number", "boolean": "Boolean"}
+
+    return converting_dict.get(schema_type, "String")
+
+
+def convert_to_xdm_type(name: str, xdm_type: str) -> str:
+    """
+    Wraps the xql with a conversion to fit the xdm schema if the raw response type is incompatible with the schema type
+    """
+    if xdm_type == "String":
+        name = to_string(name)
+    elif xdm_type == "Number":
+        name = to_number(name)
+
+    return name
+
+
+def create_xif_file(
+    mapping_list: List[MappingField], outputfile_xif: Path, dataset_name: str
+) -> None:
+    """
+    Created the xif file for the modeling rules
+    """
+    logger.info("Generating xif file\n")
+    xif_rule = create_xif_header(dataset_name)
+    for mapping_rule in mapping_list:
+        logger.info(
+            f'field name: "{mapping_rule.field_path_raw}" - xdm type: {mapping_rule.xdm_field_type} - raw type {mapping_rule.type_raw}'
+        )
+        name = mapping_rule.field_path_raw
+
+        if "." in mapping_rule.field_path_raw:
+            dict_keys = mapping_rule.field_path_raw.split(".")
+            prefix = dict_keys[0]
+            suffix = ".".join(dict_keys[1:])
+            if mapping_rule.xdm_class_type == "Array":
+                name = json_extract_array(prefix, suffix)
+                xif_rule += f"\t{mapping_rule.xdm_rule} = {name},\n"
+                continue
+            else:
+                name = json_extract_scalar(prefix, suffix)
+
+        if mapping_rule.xdm_field_type != convert_raw_type_to_xdm_type(
+            mapping_rule.type_raw
+        ):
+            # Tpue casting
+            name = convert_to_xdm_type(name, mapping_rule.xdm_field_type)
+
+        if mapping_rule.xdm_class_type == "Array":
+            # convert a scalar into an array
+            name = array_create(name)
+
+        xif_rule += f"\t{mapping_rule.xdm_rule} = {name},\n"
+
+    xif_rule = replace_last_char(xif_rule)
+
+    with open(outputfile_xif, "w") as f:
+        f.write(xif_rule)
+
+
+def replace_last_char(s: str) -> str:
+    """
+    Replaces the last char of the xif file to be ;
+    """
+    s = s[:-2]
```

Comment:
`-2` is the last two chars, why? Please clarify in docstring

Reply:
Done